### DISCUSSION TITLE
Bioschema metadata discussion in the biohackathon

### DISCUSSION THREAD

**Discussion by NishaSharma14** [NFDI4Chem/nmrxiv#679] [Data Schemas]
nmrXiv team was a part of [1st BioHackathon Germany](https://www.denbi.de/de-nbi-events/1454-biohackathon-germany) in the project of (Bio)Schemas4NFDI, lightweight domain metadata (not only) for NFDI consortia. Here you can find all the points that were solved where your comments and further improvements are welcome, along with the points that are still open for discussion.

- Solved points:
  - In an nmrXiv assay, should we add the solvent details to the dataset of the sample?
    -  This should depend on the source of information. If the solvent was extracted from the NMR files that are dataset specific, it should be added to the dataset. In case it comes from the protocol of sample preparation, it should go to the sample.
  - [Sample](https://bioschemas.org/profiles/Sample/0.2-RELEASE-2018_11_10)  will be deprecated in favor of [BioSample] (https://bioschemas.org/types/BioSample/0.1-RELEASE-2019_06_19) (BioSample is already in nmrXiv)
  - How to use [study schema](https://bioschemas.org/profiles/Study/0.3-DRAFT) for projects? 
    - This can be solved by using "conforms to" and mappingto ISA investigation.
  - In nmrXiv we have Study **about** sample, while MassBank has dataset **about** molecularEntity. We believe that both usages are fine due to the difference in the structure between nmrXiv and MassBank (no samples in MassBank)
  - Repositories in NFDI4Chem will go with the [MolecularEntity](https://bioschemas.org/types/MolecularEntity/0.3-RELEASE-2019_09_02) and will avoid [ChemicalSubstance](https://bioschemas.org/types/ChemicalSubstance/0.3-RELEASE-2019_09_02) as the first covers more molecular details and identifiers
  
- Open points:
  - How best to add the conditions/results of the assay to the dataset?
    - MassBank adds those details to "[variableMeasured](http://schema.org/variableMeasured)".
    - This solution is not ideal, as the  "[variableMeasured](http://schema.org/variableMeasured)" should refer to the target measured in the assay rather than its conditions. However, at the moment this is the best way to do it as it can have ontology terms, which is useful for the search, even if they don't fit in the right property. Bioschema organizer recommended this approach too. For the future, we aim to add more properties to the dataset, including the conditions and results. 

- Validation
  - [schema.org validator](https://validator.schema.org/)
  - [BioSchema validator](https://bioschemas-validator.herokuapp.com/) Currently down
  - [FAIR Checker]( https://fair-checker.france-bioinformatique.fr/check)

**To Join the main channels for Bioschemas, and their community.**

1. [Subscribe](mailto:public-bioschemas-request@w3.org?subject=subscribe) to the mailing list
2. Get involved in discussions on [GitHub](https://github.com/Bioschemas/specifications/issues)
3. Join the [slack space](https://join.slack.com/t/bioschemas/shared_invite/zt-1lxfiokgw-q6OENJpDv2suZrDMaNPFjg)
4. There is also a community call once a month and reminders will be sent to the mailing lists

- Tasks:
  - [ ] **[LiveDeploy to BioSchema](https://bioschemas.org/developer/liveDeploys)**
  - [ ] Get in contact with dataset group on github to inlude conditions and results
once we have the supported & validated schema, we may want to add it to live deploy page. 
https://github.com/BioSchemas/bioschemas.github.io/blob/124c4f19b272ba46859bf3c499065a1c0821630b/_data/live_deployments.json
by submitting a PR to the config file.
https://github.com/BioSchemas/bioschemas.github.io/blob/124c4f19b272ba46859bf3c499065a1c0821630b/_data/live_deployments.json
  - [ ] **BioSchema Markup**
Rene implemented this in https://github.com/MassBank/MassBank-web/commit/44790c61edaeec2f81dc2ea4d9651b561e7d5143
using the Java GSON library for proper encoding. For PHP there could be the equivalent https://github.com/tebru/gson-php
  - DataCatalog:
    - [ ] Add version
  - Project
    - [x] conform to ISA
  - Study
    - [x] conform to ISA
    - [x] Re-define the structure the output json. May be the study details can be displayed first followed by project and dataset details. May be `isPartOf` is something we can use.
  -  [Dataset](https://bioschemas.org/profiles/Dataset/1.0-RELEASE)
    - [x] use  "[variableMeasured](http://schema.org/variableMeasured)"  for NMRium info and results
    - [x]  Add "[distribution](http://schema.org/distribution)" when available 

- Future tasks:
  - [ ]  Add "[distribution](http://schema.org/distribution)" when available
  - [ ]  Add version when available

**Comment by NishaSharma14** [comment]
**To look for json schemas:**

1. Go to https://discovery.biothings.io/registry
2. Go to the tab "Browse by Namespace"
3. If you are looking for a release you can go to "Bioschemas" (https://discovery.biothings.io/view/bioschemas)
4. If you are looking for a draft you ca go to "BioschemasDrafts" (https://discovery.biothings.io/view/bioschemasdrafts)
5. Find the profile you are interested in, for instance Dataset (released version at https://discovery.biothings.io/view/bioschemas/bioschemas:Dataset) and navigate to the page
6. On the top right, there is a link "View Source"

**For markup examples**

1. Go to https://bioschemas.org/
2. Navigate to the profiles list from the menu Specifications-Profiles (https://bioschemas.org/profiles/)
3. Stay on the tab "Release" or go to the tab "Draft" depending on the profile you are interested in
4. For instance, if you are interested in Dataset, you would end up at https://bioschemas.org/profiles/Dataset
5. On the top, go to the tab "Links" (https://bioschemas.org/profiles/Dataset/1.0-RELEASE#nav-links)
6. Look for the "Examples" link and there you will find some examples, commonly corresponding to the minimum properties

Alternative

1. Go to Live Deploys and navigate by Profile (https://bioschemas.org/developer/liveDeploys#nav-profile)
2. Find the profile you are interested in
3. Open the page for any of the listed resources
4. Use Developer Tools in your browser and locate the piece in the HTML corresponding to a script "application/ld+json"
5. You should get the JSON-LD markup which you can copy to an editor to have a better look

**Comment by NRayya** [comment]
Most of the decided features are already provided. Others will be discussed more in the TA15 regular meetings
